crates/rpc/rpc/src/eth/helpers/types.rs

```diff
@@ -0,0 +1,85 @@
+//! L1 `eth` API types.
+
+use alloy_network::{AnyNetwork, Network};
+use reth_primitives::{Address, TransactionSignedEcRecovered, TxKind};
+use reth_rpc_types::{Transaction, TransactionInfo, WithOtherFields};
+use reth_rpc_types_compat::{
+    transaction::{from_primitive_signature, GasPrice},
+    TransactionCompat,
+};
+
+/// Builds RPC transaction response for l1.
+#[derive(Debug, Clone, Copy)]
+pub struct EthTxBuilder;
+
+impl TransactionCompat for EthTxBuilder
+where
+    Self: Send + Sync,
+{
+    type Transaction = <AnyNetwork as Network>::TransactionResponse;
+
+    fn fill(tx: TransactionSignedEcRecovered, tx_info: TransactionInfo) -> Self::Transaction {
+        let signer = tx.signer();
+        let signed_tx = tx.into_signed();
+
+        let to: Option<Address> = match signed_tx.kind() {
+            TxKind::Create => None,
+            TxKind::Call(to) => Some(Address(*to)),
+        };
+
+        let TransactionInfo {
+            base_fee, block_hash, block_number, index: transaction_index, ..
+        } = tx_info;
+
+        let GasPrice { gas_price, max_fee_per_gas } =
+            Self::gas_price(&signed_tx, base_fee.map(|fee| fee as u64));
+
+        let chain_id = signed_tx.chain_id();
+        let blob_versioned_hashes = signed_tx.blob_versioned_hashes();
+        let access_list = signed_tx.access_list().cloned();
+        let authorization_list = signed_tx.authorization_list().map(|l| l.to_vec());
+
+        let signature = from_primitive_signature(
+            *signed_tx.signature(),
+            signed_tx.tx_type(),
+            signed_tx.chain_id(),
+        );
+
+        WithOtherFields {
```

Comment:
do we still nicht WithOtherFields here?

Reply:
when I implemented this is it wasn't possible to use `Ethereum` instead of `AnyNetwork`, may have changed now, though appreciate if this can be tackled in a follow pr, will open issue